spec/models/reports/facility_state_spec.rb

```diff
@@ -259,6 +259,232 @@
     end
   end
 
+  context "Diabetes" do
+    context "registraions" do
+      describe "cumulative_diabetes_registrations" do
+        it "has the total number of diabetic registrations since the starting of time for every facility upto a given month" do
+          facility = create(:facility)
+          old_diabetes_patients = create_list(:patient, 2, :diabetes, registration_facility: facility, recorded_at: june_2021[:long_ago])
+          new_diabetic_patients = create_list(:patient, 3, :diabetes, registration_facility: facility, recorded_at: june_2021[:two_months_ago])
+          _non_diabetic_patients = create_list(:patient, 3, :without_diabetes, registration_facility: facility, recorded_at: june_2021[:two_months_ago])
+
+          refresh_views
+          with_reporting_time_zone do
+            expect(described_class
+                     .where(facility_id: facility.id)
+                     .where("month_date < ?", june_2021[:long_ago])
+                     .pluck(:cumulative_diabetes_registrations)).to all eq 0
+
+            expect(described_class
+                     .where(facility_id: facility.id)
+                     .where("month_date >= ?", june_2021[:long_ago])
+                     .where("month_date < ?", june_2021[:two_months_ago])
+                     .pluck(:cumulative_diabetes_registrations)).to all eq old_diabetes_patients.size
+
+            expect(described_class
+                     .where(facility_id: facility.id)
+                     .where("month_date >= ?", june_2021[:two_months_ago])
+                     .pluck(:cumulative_diabetes_registrations)).to all eq (old_diabetes_patients + new_diabetic_patients).size
+          end
+        end
+      end
+
+      describe "monthly_diabetes_registrations" do
+        it "has the number of diabetes patients registered in a given month" do
+          facility = create(:facility)
+          old_diabetes_patients = create_list(:patient, 2, :diabetes, registration_facility: facility, recorded_at: june_2021[:under_12_months_ago])
+          new_diabetic_patients = create_list(:patient, 3, :diabetes, registration_facility: facility, recorded_at: june_2021[:two_months_ago])
+          _non_diabetic_patients = create_list(:patient, 3, :without_diabetes, registration_facility: facility, recorded_at: june_2021[:two_months_ago])
+
+          month_2020_07 = "2020-07-01"
+          month_2021_04 = "2021-04-01"
+
+          refresh_views
+          with_reporting_time_zone do
+            expect(described_class
+                     .where(facility_id: facility.id)
+                     .where("month_date < ?", month_2020_07)
```

Comment:
Date comparisons always make me nervous with timezone handling etc. Can we either confirm that this assertion is doing what we want, or can we instead compare with `(month: 7, year: 2020)`?

Reply:
I'd like to use the `month_date` column here since that's what we mostly use in our pipeline downstream. I've confirmed that the query produces the required results.

Reply:
Neato, thanks.